src/Microsoft.TestPlatform.Extensions.BlameDataCollector/ProcessDumpUtility.cs

```diff
@@ -100,27 +99,38 @@ private static string BuildProcDumpArgs(int processId, string filename)
             // -accepteula: Auto accept end-user license agreement
             // -t: Write a dump when the process terminates.
             // This will create a minidump of the process with specified filename
-            return "-accepteula -t " + processId + " " + filename + ".dmp";
+            return "-accepteula -e 1 -g -t -f STACK_OVERFLOW -f ACCESS_VIOLATION " + processId + " " + filename + ".dmp";
         }
 
         /// <summary>
         /// Get procdump executable path
         /// </summary>
+        /// <param name="processId">
+        /// Process Id
+        /// </param>
         /// <returns>procdump executable path</returns>
-        private string GetProcDumpExecutable()
+        private string GetProcDumpExecutable(int processId)
         {
             var procdumpPath = Environment.GetEnvironmentVariable("PROCDUMP_PATH");
+            string processName = this.processHelper.GetProcessName(processId);
+
+            if (EqtTrace.IsVerboseEnabled)
+            {
+                EqtTrace.Verbose("Using process name {0}", processName);
+            }
+
             if (!string.IsNullOrWhiteSpace(procdumpPath))
             {
                 string filename = string.Empty;
 
-                if (this.environment.Architecture == PlatformArchitecture.X64)
+                // Launch procdump according to process architecture
+                if (this.environment.Architecture == PlatformArchitecture.X86 || processName == Constants.X86TestHostProcessName)
```

Comment:
ProcDumpUtility should not be aware of what process it is taking dump. So dont use `X86TestHostProcessName` name directly here. Instead pass it via blameCollector.

Reply:
Using `IsWow64`.